packages/amplify-e2e-core/src/utils/add-circleci-tags.ts

```diff
@@ -12,8 +14,13 @@ declare global {
   /* eslint-enable */
 }
 
-export const addCircleCITags = (projectPath: string): void => {
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+export const addCircleCITags = async (projectPath: string): Promise<void> => {
   if (process.env && process.env['CIRCLECI']) {
+    await staggerTestsThatRunOnTheSameMachine(projectPath);
```

Comment:
This is basic round robin scheduling during test startup.

I implemented it using the OS/filesystem because there is no other way to communicate between tests, and running init multiple times concurrently seems to cause a lot of errors with simulators (they apparently create files, and throw error if someone else has created them)